agent/discover/discover.go

```diff
@@ -24,10 +25,15 @@ type Discoverer interface {
 }
 
 type Discover struct {
+	Id     string
```

Comment:
nit: since we have the `GetId()` method, I think we can make this private.

Reply:
Yes...
In any case, this id is only used a the consul `checkId`, and I'm not sure if we should integrate the discovery mechanisms as a consul check